File: python/FactoryMethod/complete/MusicProduct.py
from Product import Product

class MusicProduct(Product):

    def is_equal(self, name):
        '''
        returns true if the name matches this products name - case insensitive
        '''
        return self.name.lower() == name.lower()

    def get_string(self):
        return "Music : "+self.name

File: java/FactoryMethod/complete/MusicProduct.java
class MusicProduct extends Product{

  /*MusicProduct constructor, same functionality as Product constructor*/
  public MusicProduct(String name){
    super(name);
  }

  /*returns true if the name matches this products name - case insensitive*/
  public Boolean is_equal(String name){
    return this.name.toLowerCase().equals(name.toLowerCase());
  }

  public String get_string(){
    return "Music : "+this.name;
  }
}
